### DISCUSSION TITLE
Code style ideas

### DISCUSSION THREAD

**Discussion by zhuiks** [ezra-bible-app/ezra-bible-app#255] [Development]
Let's discuss code style practices. Maybe later it can become a part of the project documentation

**Comment by zhuiks** [comment]
## var, let, const

I think the most important is to define on literal constants (strings and numbers) with `const` and define inner scope variables (such as used only inside the loop or if clause) with `let` or `const`. Everything else has just semantic meaning to improve readability of the code.

Some ideas:
- `var` can display that variable is **"global" for module or function scope**
- `const` can display that variable is **assigned only once during its definition**. We have to be careful as objects and arrays defined with `const` are **not immutable** :exclamation: (you can change object properties and array elements)

**Reply by tobias-klein** [reply]
Thanks for the proposal. I'll have a look and consider this next weekend!
